scripts/postlink/ios/postlink.js

```diff
@@ -94,3 +98,42 @@ function findFileByAppName(array, appName) {
 
     return null;
 }
+
+function getDefaultPlistPath() {
+    //this is old logic in case we are unable to find PLIST from xcode/pbxproj - at least we can fallback to default solution
+    return glob.sync(`**/${package.name}/*Info.plist`, ignoreNodeModules)[0];
+}
+
+function getPlistPath(){
+    var xcodeProjectPaths = glob.sync(`**/*.xcodeproj/project.pbxproj`, ignoreNodeModules);
+    if (!xcodeProjectPaths){
+        return getDefaultPlistPath();
+    }
+
+    if (xcodeProjectPaths.length !== 1) {
+        console.log('Could not determine correct xcode proj path to retrieve related plist file, there are multiple xcodeproj under the solution.');
+        return getDefaultPlistPath();
+    }
+
+    var xcodeProjectPath = xcodeProjectPaths[0];
+    var parsedXCodeProj;
+
+    try {
+        var proj = xcode.project(xcodeProjectPath);      
+        //use sync version because there are some problems with async version of xcode lib as of current version
+        parsedXCodeProj = proj.parseSync();
+    }
+    catch(e) {
+        console.log('Couldn\'t read info.plist path from xcode project - error: ' + e.message);
+        return getDefaultPlistPath();
+    }
+
+    var INFO_PLIST_PROJECT_KEY = 'INFOPLIST_FILE';
+    var plistPathValue = parsedXCodeProj.getBuildProperty(INFO_PLIST_PROJECT_KEY);
```

Comment:
@max-mironov that's a great point. And yep, that sounds like a good solution. Between that and specifying the build type to `getBuildProperty()` I think it would work quite well :)